### ISSUE TITLE
  EnableRemoteAccess = true + EnableUPnP =true causes immediate crash

### ISSUE THREAD

**Issue by aljazzair** [jellyfin/jellyfin#5148]
**Describe the bug**
Selecting EnableRemoteAccess = true + EnableUPnP =true causes immediate crash of server upon startup.

**System (please complete the following information):**
 - OS: Manjaro/Arch
`Linux server 5.4.89-1-MANJARO #1 SMP PREEMPT Tue Jan 12 23:39:44 UTC 2021 x86_64 GNU/Linux`
 - Virtualization: KVM
```
 qemu-system-x86_64 --version
QEMU emulator version 5.2.0
```

 - Clients: Not Relevant
 - Browser:Not Relevant
 - Jellyfin Version: Latest from AUR
```
 jellyfin-web-bin                    [Installed] 10.7.0~rc3-1              AUR 
    The Free Software Media System - web package
jellyfin-server-bin                  [Installed] 10.7.0~rc3-1              AUR 

```
 - Playback: Not relevant
 - Installed Plugins:  Several but does not seem relevant
 - Reverse Proxy:None
 - Base URL: https://1.2.3.4:2020/jellyfin
 - Networking: lo , enp1s0 (local), tun0 (vpn)
```
 enp1s0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.30.1  netmask 255.255.255.0  
        ether aa:xx:zz:zz:zz:zz  txqueuelen 1000  (Ethernet)
        RX packets 35503  bytes 3621954 (3.4 MiB)
        RX errors 0  dropped 4354  overruns 0  frame 0
        TX packets 28426  bytes 78566642 (74.9 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 2041  bytes 146016 (142.5 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2041  bytes 146016 (142.5 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

tun0: flags=4305<UP,POINTOPOINT,RUNNING,NOARP,MULTICAST>  mtu 1500
        inet 192.168.55.44  netmask 255.255.255.0  
        inet6 fe80::cxx:6xx9:8xxx:5xxx  prefixlen 64  scopeid 0x20<link>
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 500  (UNSPEC)
        RX packets 1959  bytes 696995 (680.6 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2027  bytes 154251 (150.6 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

```
 - Storage: local (zfs)

**To Reproduce**
Installed Jellyfin from AUR, added library, everything works smoothly when Remote Access or UPNP is disabled.
As soon as Remote Access and UPNP are enabled simultaneously, the server crashes upon restart.
1. enable Remote access and UPNP through GUI or network.xml
2. sudo systemctl restart jellyfin.service
3. Crash

**Expected behavior**
**Either Enabling Remote access and UPNP should not crash server or it should not be possible to enable both simultaneously.**

**Logs**
Partial Log from /var/log/jellyfin/2021-02-xyz.log
```
[2021-02-01 21:58:48.741 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: ServerId: "a4bb6bf7db8140bfab1cd78162b2a2c5"
[2021-02-01 21:58:48.745 +01:00] [ERR] [1] Emby.Dlna.Main.DlnaEntryPoint: The DLNA specification does not support HTTPS.
[2021-02-01 21:58:48.830 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Executed all pre-startup entry points in 0:00:00.0793251
[2021-02-01 21:58:48.830 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Core startup complete
[2021-02-01 21:58:48.965 +01:00] [INF] [1] Emby.Server.Implementations.EntryPoints.ExternalPortForwarding: Starting NAT discovery
[2021-02-01 21:58:48.971 +01:00] [INF] [10] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory /drive/vids/funny
[2021-02-01 21:58:48.977 +01:00] [INF] [8] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory /var/lib/jellyfin/data/collections
[2021-02-01 21:58:49.024 +01:00] [INF] [11] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory /drive/youtube/funny
[2021-02-01 21:58:49.052 +01:00] [FTL] [1] Main: Error while starting server.
System.ArgumentNullException: Value cannot be null. (Parameter 'communicationsServer')
   at Rssdp.Infrastructure.SsdpDeviceLocator..ctor(ISsdpCommunicationsServer communicationsServer)
   at Emby.Dlna.Ssdp.DeviceDiscovery.StartInternal()
   at Emby.Dlna.Ssdp.DeviceDiscovery.add_DeviceDiscovered(EventHandler`1 value)
   at Emby.Server.Implementations.EntryPoints.ExternalPortForwarding.Start()
   at Emby.Server.Implementations.EntryPoints.ExternalPortForwarding.RunAsync()
   at Emby.Server.Implementations.ApplicationHost.StartEntryPoints(IEnumerable`1 entryPoints, Boolean isBeforeStartup)+MoveNext()
   at System.Threading.Tasks.Task.WhenAll(IEnumerable`1 tasks)
   at Emby.Server.Implementations.ApplicationHost.RunStartupTasksAsync()
   at Jellyfin.Server.Program.StartApp(StartupOptions options)
[2021-02-01 21:58:49.076 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "CoreAppHost"
[2021-02-01 21:58:49.076 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "EmbyTV"
[2021-02-01 21:58:49.077 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "TMDbBoxSetManager"
[2021-02-01 21:58:49.078 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "EntryPoint"
[2021-02-01 21:58:49.078 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "LibraryMonitorStartup"
[2021-02-01 21:58:49.078 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "ExternalPortForwarding"
[2021-02-01 21:58:49.079 +01:00] [INF] [1] Emby.Server.Implementations.EntryPoints.ExternalPortForwarding: Stopping NAT discovery
[2021-02-01 21:58:49.084 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "LibraryChangedNotifier"
[2021-02-01 21:58:49.086 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "RecordingNotifier"
[2021-02-01 21:58:49.087 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "UdpServerEntryPoint"
[2021-02-01 21:58:49.087 +01:00] [ERR] [1] Emby.Server.Implementations.ApplicationHost: Error disposing "UdpServerEntryPoint"
System.NullReferenceException: Object reference not set to an instance of an object.
   at Emby.Server.Implementations.EntryPoints.UdpServerEntryPoint.Dispose()
   at Emby.Server.Implementations.ApplicationHost.Dispose(Boolean dispose)
[2021-02-01 21:58:49.088 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "UserDataChangeNotifier"
[2021-02-01 21:58:49.088 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "DlnaEntryPoint"
[2021-02-01 21:58:49.089 +01:00] [INF] [1] Emby.Dlna.Main.DlnaEntryPoint: Disposing DeviceDiscovery
[2021-02-01 21:58:49.089 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "NotificationEntryPoint"
[2021-02-01 21:58:49.091 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "EntryPoint"
[2021-02-01 21:58:49.091 +01:00] [INF] [1] Emby.Server.Implementations.ApplicationHost: Disposing "DeviceAccessEntryPoint"
[2021-02-01 21:58:49.093 +01:00] [INF] [2] Main: Received a SIGTERM signal, shutting down
```

**Screenshots**

**Additional context** (not sure if it's relevant)
using intelGVT-g for ffmpeg decoding
running jellyfin as specific user username/group, all relevant file ownership changed accordingly
```
sudo systemctl edit jellyfin.service
[Service]
User=username
Group=groupname
```
++
sudo find / -user jellyfin and chown accordingly

**Comment by crobibero** [comment]
I can't replicate this on my development server running rc3

**Comment by RoroTiti** [comment]
I had the same issue, but after multiple server reboots, it finally started to boot... Strange issue (using RC3).

**Comment by BaronGreenback** [comment]
Looks like the speed improvements to 10.7 means the code is now detecting devices before the main system is ready to handle them - hence the null argument error. Will submit a pr

**Comment by Flexusma** [comment]
This issue seems to reappear on stable jellyfin 10.7.0. Using virtualized Debian 10 (kernel 4.19.0-10). I tried disabling UPnP but i keep getting the same message 
![image](https://user-images.githubusercontent.com/38422530/110683389-3eb88380-81dc-11eb-9ea3-cbb94192549b.png)

**Comment by jairbubbles** [comment]
Same on my side, just updated and I got the same error. (using an ARM Qnap NAS)

**Comment by Flexusma** [comment]
@jairbubbles Do you also see [ERR] The DLNA specification does not support HTTPS. ? Seems to be some problem with DLNA am trying to figure it out rn

**Comment by BaronGreenback** [comment]
Same cause - different bug. I'll submit a PR for backporting.

**Comment by BaronGreenback** [comment]
Same cause as before - different bug. I'll submit a PR for backporting.

**Comment by jairbubbles** [comment]
Callstack is:
```
[19:17:31] [FTL] [1] Main: Error while starting server.,
System.ArgumentNullException: Value cannot be null. (Parameter 'communicationsServer'),
   at Rssdp.Infrastructure.SsdpDeviceLocator..ctor(ISsdpCommunicationsServer communicationsServer),
   at Emby.Dlna.Ssdp.DeviceDiscovery.StartInternal(),
   at Emby.Dlna.Ssdp.DeviceDiscovery.add_DeviceDiscovered(EventHandler`1 value),
   at Emby.Server.Implementations.EntryPoints.ExternalPortForwarding.Start(),
   at Emby.Server.Implementations.EntryPoints.ExternalPortForwarding.RunAsync(),
   at Emby.Server.Implementations.ApplicationHost.StartEntryPoints(IEnumerable`1 entryPoints, Boolean isBeforeStartup)+MoveNext(),
   at System.Threading.Tasks.Task.WhenAll(IEnumerable`1 tasks),
   at Emby.Server.Implementations.ApplicationHost.RunStartupTasksAsync(),
   at Jellyfin.Server.Program.StartApp(StartupOptions options),
```

@Flexusma Yes I see just before:
```
[19:17:30] [ERR] [1] Emby.Dlna.Main.DlnaEntryPoint: The DLNA specification does not support HTTPS.,
```

**Comment by BaronGreenback** [comment]
DLNA v1.1 does not support https in the url part of the response in the ssdp spec.

I'll be testing DLNA v2 shortly as there is something in the spec about https there.

**Comment by Flexusma** [comment]
The thing is, I've tried to disable DLNA as well as resetting my config but for some reason I still get the HTTPS error :/

**Comment by Flexusma** [comment]
Found a FIX @BaronGreenback. 

- Make sure to set ForceHTTPS to false in every config (network.xml & system.xml)
currently seeems like UPnP crashes when it is forced to bind to https connection/port

**Comment by BaronGreenback** [comment]
DLNA still part of the main system, and as such is initialised in 10.7. The dlna subsystem is disabled in 10.7, and it looks like dlna traffic is being recieved before initialisation is complete - which is cause of null pointer.
I'm currently working on moving it into a plugin.

I'm currently testing 10.X which has dlna as plugins.

Could you try PR https://github.com/jellyfin/jellyfin/pull/5475 as this should fix the null pointer.

**Comment by jairbubbles** [comment]
@BaronGreenback  Is the PR's CI producing docker images that can be pulled?

**Comment by crobibero** [comment]
> @BaronGreenback Is the PR's CI producing docker images that can be pulled?

PRs do not create images, sorry

**Comment by jairbubbles** [comment]
Does `add_DeviceDiscovered` indicates that a DLNA server is discovered on my local network? I'm trying to find the culprit with no luck until now.

**Comment by BaronGreenback** [comment]
with dlna disabled - deviceDiscovered **shouldn't** be run.
The dlna code is a bit of a maze which is part of the reason it's being moved into a plugin.

https://github.com/jellyfin/jellyfin/pull/4543

**Comment by jairbubbles** [comment]
By looking closely at the code of PR #5475, I can say with confidence it will fix the null reference exception.

To make sure there is no other issue hindering it would need to be tested in my live environment but it's and Arm device and I'm using the docker images so it's not the best setup to play with. @Flexusma  are you able to test the PR on your side?

**Comment by Flexusma** [comment]
@jairbubbles I fixed it for me by disabling forceHTTPS, so there is technically no need for me to further "fix" this issue. Have you tried that yet? Might be a viable option until the PR is in master

**Comment by jairbubbles** [comment]
@Flexusma  Yep thanks for the tip, the option was named `<RequireHttps>` on my side but it did the trick!

**Comment by BaronGreenback** [comment]
Fixed with #5475
